### DISCUSSION TITLE
[Feature Request] Endpoint caching

### DISCUSSION THREAD

**Discussion by BulgarianHealer** [tailflow/laravel-orion#300] [Ideas]
It will be awesome if there is a default cache or customizable cache. 

I was thinking do add it, but I'm not sure if using middleware is gonna be the best for this package or creating custom controllers, configs.

Also if there is a cache option it should be enabled as default, and disable as DisableCache trait.

**Comment by PascalHesselink** [comment]
There's an awesome package that does this exactly for you;
https://github.com/GeneaLabs/laravel-model-caching

On topic: I don't think a cache should be enabled by default

**Comment by BulgarianHealer** [comment]
> There's an awesome package that does this exactly for you;
> https://github.com/GeneaLabs/laravel-model-caching
> 
> On topic: I don't think a cache should be enabled by default

I haven't seen this package I will give it a try to see if is compatible with this package.
Edit:  This package will not work with laravel-orion(custom query build)

On the topic:
Well let's say that there is an app that is live, but there is some little bug, so you turn on debug mode, and the cache will be disabled. If the app is set local env, not to the prod. it will be disabled.
The other thing is that when we/you/they add more packages it's getting hard and hard to upgrade everything to the latest version of laravel. I have seen apps with more than 60 packages and some of them were added as a missing feature of some packages(that app is using). And after some time maybe the developer will stop to support it and you know what is happening. 
So I think maybe will be better for this package to have a standalone cacheable trait.

**Comment by alexzarbn** [comment]
Hi @BulgarianHealer,

Yes, caching is something I was considering to add. As you also mentioned, there are quite a few ways to do that, and to be honest, I do not have a clear vision (yet) on how to implement this feature either. 

What makes it difficult is cache revalidation, that should work hand in hand with other (possible) cache implementations of the apps that use both Laravel Orion and third-party caching packages. One thing that comes in mind is to have its own implementation of caching and bridges for those packages (e.g. `LaravelModelCachingBridge`) to ensure proper revalidation, should an app require both implementations.
